### PR TITLE
TLS: converge asset naming of SH and non-SH etcd

### PR DESCRIPTION
**Pull request by hongchaodeng**
fix https://github.com/kubernetes-incubator/bootkube/issues/611

### CODE REVIEW

**Comment by hongchaodeng** [comment]
Manually verified recovery case too.

**Comment by hongchaodeng** [comment]
@xiang90 @diegs @aaronlevy 
Can you take a review?

**Comment by hongchaodeng** [comment]
coreosbot run e2e

Seems flake:
```
reboot_test.go:44: rebooting node: ssh: handshake failed: EOF
```

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
@aaronlevy @diegs Can you approve this?

File: hack/multi-node/etcd-cloud-config.yaml [hunk 1]
```diff
@@ -21,10 +21,10 @@ coreos:
           Environment="ETCD_LISTEN_PEER_URLS=https://$private_ipv4:2380"
           Environment="ETCD_INITIAL_CLUSTER={{ETCD_INITIAL_CLUSTER}}"
           Environment="ETCD_SSL_DIR=/etc/etcd/tls"
-          Environment="ETCD_TRUSTED_CA_FILE=/etc/ssl/certs/etcd-ca.crt"
+          Environment="ETCD_TRUSTED_CA_FILE=/etc/ssl/certs/etcd-client-ca.crt"
```
Comment (diegs) [inline hack/multi-node/etcd-cloud-config.yaml]: etcd-server-ca.rt?

File: hack/multi-node/etcd-cloud-config.yaml [hunk 2]
```diff
@@ -21,10 +21,10 @@ coreos:
...
           Environment="ETCD_CERT_FILE=/etc/ssl/certs/etcd-client.crt"
           Environment="ETCD_KEY_FILE=/etc/ssl/certs/etcd-client.key"
           Environment="ETCD_CLIENT_CERT_AUTH=true"
-          Environment="ETCD_PEER_TRUSTED_CA_FILE=/etc/ssl/certs/etcd-ca.crt"
-          Environment="ETCD_PEER_CERT_FILE=/etc/ssl/certs/etcd-peer.crt"
-          Environment="ETCD_PEER_KEY_FILE=/etc/ssl/certs/etcd-peer.key"
+          Environment="ETCD_PEER_TRUSTED_CA_FILE=/etc/ssl/certs/etcd-client-ca.crt"
```
Comment (diegs) [inline hack/multi-node/etcd-cloud-config.yaml]: shouldn't this be `peer-ca.crt`?

File: hack/single-node/user-data-etcd.sample [hunk 3]
```diff
@@ -12,11 +12,11 @@
...
             Environment="ETCD_SSL_DIR=/etc/etcd/tls"
-            Environment="ETCD_TRUSTED_CA_FILE=/etc/ssl/certs/etcd-ca.crt"
+            Environment="ETCD_TRUSTED_CA_FILE=/etc/ssl/certs/etcd-client-ca.crt"
             Environment="ETCD_CERT_FILE=/etc/ssl/certs/etcd-client.crt"
             Environment="ETCD_KEY_FILE=/etc/ssl/certs/etcd-client.key"
             Environment="ETCD_CLIENT_CERT_AUTH=true"
-            Environment="ETCD_PEER_TRUSTED_CA_FILE=/etc/ssl/certs/etcd-ca.crt"
-            Environment="ETCD_PEER_CERT_FILE=/etc/ssl/certs/etcd-peer.crt"
-            Environment="ETCD_PEER_KEY_FILE=/etc/ssl/certs/etcd-peer.key"
+            Environment="ETCD_PEER_TRUSTED_CA_FILE=/etc/ssl/certs/etcd-client-ca.crt"
```
Comment (diegs) [inline hack/single-node/user-data-etcd.sample]: etcd-peer-ca.crt?

File: hack/quickstart/init-master.sh [hunk 4]
```diff
@@ -34,13 +36,13 @@ Environment="ETCD_ADVERTISE_CLIENT_URLS=https://${COREOS_PRIVATE_IPV4}:2379"
...
 Environment="ETCD_SSL_DIR=/etc/etcd/tls"
-Environment="ETCD_TRUSTED_CA_FILE=/etc/ssl/certs/etcd-ca.crt"
+Environment="ETCD_TRUSTED_CA_FILE=/etc/ssl/certs/etcd-client-ca.crt"
 Environment="ETCD_CERT_FILE=/etc/ssl/certs/etcd-client.crt"
 Environment="ETCD_KEY_FILE=/etc/ssl/certs/etcd-client.key"
 Environment="ETCD_CLIENT_CERT_AUTH=true"
-Environment="ETCD_PEER_TRUSTED_CA_FILE=/etc/ssl/certs/etcd-ca.crt"
-Environment="ETCD_PEER_CERT_FILE=/etc/ssl/certs/etcd-peer.crt"
-Environment="ETCD_PEER_KEY_FILE=/etc/ssl/certs/etcd-peer.key"
+Environment="ETCD_PEER_TRUSTED_CA_FILE=/etc/ssl/certs/etcd-client-ca.crt"
```
Comment (diegs) [inline hack/quickstart/init-master.sh]: peer-ca.crt?

File: hack/single-node/user-data-etcd.sample [hunk 5]
```diff
@@ -12,11 +12,11 @@
             Environment="ETCD_LISTEN_CLIENT_URLS=https://0.0.0.0:2379"
             Environment="ETCD_LISTEN_PEER_URLS=https://0.0.0.0:2380"
             Environment="ETCD_SSL_DIR=/etc/etcd/tls"
-            Environment="ETCD_TRUSTED_CA_FILE=/etc/ssl/certs/etcd-ca.crt"
+            Environment="ETCD_TRUSTED_CA_FILE=/etc/ssl/certs/etcd-client-ca.crt"
```
Comment (diegs) [inline hack/single-node/user-data-etcd.sample]: etcd-server-ca.crt?

**Comment by hongchaodeng** [comment]
@diegs 
Let me answer all of the questions above.
So those cloud config yamls are for non-self-hosted cases.
In current setup, non-self-hosted assets for etcd includes:
```
etcd-client-ca.crt
etcd-client.[crt,key]
etcd-peer.[crt,key]
```

**Comment by diegs** [comment]
@hongchaodeng couldn't the non-self-hosted code path in `bootkube render` also output `server-ca.crt` `peer-ca.crt` too (even if they're both copies of `etcd-client-ca.crt`) to make things consistent and clear?

(Discussed in 2 comments at https://github.com/kubernetes-incubator/bootkube/issues/611#issuecomment-311134671, I know it's a long thread so thanks for helping to clean all this up)

**Comment by hongchaodeng** [comment]
@diegs 
Could we do it in another PR? This PR is already huge. And this PR is only about converging the paths.

File: pkg/asset/tls.go [hunk 6]
```diff
@@ -139,7 +139,7 @@ func newEtcdTLSAssets(etcdCACert, etcdClientCert *x509.Certificate, etcdClientKe
 	}
 
 	assets = append(assets, []Asset{
-		{Name: AssetPathEtcdCA, Data: tlsutil.EncodeCertificatePEM(etcdCACert)},
+		{Name: AssetPathEtcdClientCA, Data: tlsutil.EncodeCertificatePEM(etcdCACert)},
```
Comment (diegs) [inline pkg/asset/tls.go]: Please add `AssetPathEtcdServer[CA,Key,Cert]` here.

Please add `AssetPathEtcdPeerCA` on line 136 above.

File: hack/multi-node/etcd-cloud-config.yaml (hunk 1 above)
Comment (diegs) [inline hack/multi-node/etcd-cloud-config.yaml]: (and the below should be `etcd-server.crt` and `etcd-server.key`

File: hack/single-node/user-data-etcd.sample (hunk 5 above)
Comment (diegs) [inline hack/single-node/user-data-etcd.sample]: (and the two lines below)

**Comment by hongchaodeng** [comment]
All fixed

**Comment by xiang90** [comment]
coreosbot run e2e

**Pull request merged**